src/modules/pair/services/pair.metadata.builder.ts

```diff
@@ -1,20 +1,25 @@
-import { PairsFilter } from 'src/modules/router/models/filter.args';
+import {
+    PairFilterArgs,
+    PairsFilter,
+} from 'src/modules/router/models/filter.args';
 import { PairMetadata } from 'src/modules/router/models/pair.metadata.model';
+import { PairModel } from '../models/pair.model';
+import { PairService } from './pair.service';
+import { PairComputeService } from './pair.compute.service';
+import { EsdtToken } from 'src/modules/tokens/models/esdtToken.model';
 import { PairFilteringService } from './pair.filtering.service';
 
 export class PairsMetadataBuilder {
-    private pairsMetadata: PairMetadata[];
-    private filters: PairsFilter;
-    private filteringService: PairFilteringService;
+    private pairs: PairModel[];
+    private filters: PairsFilter | PairFilterArgs;
+    private pairService: PairService;
+    private pairCompute: PairComputeService;
+    private lpTokensIndexed: boolean;
 
-    constructor(
-        pairsMetadata: PairMetadata[],
-        filters: PairsFilter,
-        filteringService: PairFilteringService,
-    ) {
-        this.pairsMetadata = pairsMetadata;
-        this.filters = filters;
-        this.filteringService = filteringService;
+    constructor(pairService: PairService, pairCompute: PairComputeService) {
```

Comment:
I think we can inject all the services in the constructor and also inject the builder class where it is used